### PR TITLE
Check in/70311/upcoming details link

### PR DESCRIPTION
**Pull request by brianseek**
## Summary

Adds the ability to use the details page component on upcoming appointments. Anything in the normal appointments array supersedes rules on upcoming appointments. Appointments that are only in the upcoming array do not have a message or action.

Also the reload wrapper will now fetch upcoming appointments to allow upcoming details to be reloadable. 

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#70311

## Testing done

- Updates unit test

## What areas of the site does it impact?

Unified check-in

## Acceptance criteria

[ ] - Details links for for upcoming appointments
[ ] - You can reload the details page

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### CODE REVIEW

**Review by lee-delarm6 — APPROVED** [review]
It's looking good to me!

File: src/applications/check-in/components/pages/AppointmentDetails/index.jsx [hunk 1]
```diff
@@ -125,12 +135,18 @@ const AppointmentDetails = props => {
                     : t('in-person-appointment')
                 }`}
               </h1>
-              {app === APP_NAMES.PRE_CHECK_IN ? (
-                preCheckInSubTitle
+              {isUpcoming ? (
```
Comment (wullaski) [inline src/applications/check-in/components/pages/AppointmentDetails/index.jsx:138]: Could probably change this to `{!isUpcoming && (...)`

**Review by wullaski — APPROVED** [review]
Just a small nit pick but looks great!

File: src/applications/check-in/components/pages/AppointmentDetails/index.jsx (hunk 1 above)
Comment (lee-delarm6) [reply src/applications/check-in/components/pages/AppointmentDetails/index.jsx:138]: I thought about that, but I was like, nah, maybe he wants to add something here in the future

Comment (brianseek) [reply src/applications/check-in/components/pages/AppointmentDetails/index.jsx]: I fixed it. I think my brain defaults to this syntax for some reason. But thanks for the benefit of the doubt there Lee!

**Pull request merged**